packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -71,6 +93,37 @@ export class GOVUKFrontendComponent {
 
     this.checkInitialised()
 
+    if (configObjects.length) {
+      const childConstructorWithConfig =
+        /** @type {ChildClassConstructorConfig<ConfigType>} */ (
+          this.constructor
+        )
+
+      const dataset = /** @type {{ dataset: DOMStringMap }} */ (
+        /** @type {unknown} */ (this._$root)
+      ).dataset
+      const normalisedDataset = /** @type {ConfigType} */ (
+        normaliseDataset(childConstructorWithConfig, dataset)
+      )
+
+      const configObjectOveride = /** @type {ConfigType} */ (
+        childConstructorWithConfig.configOverride(normalisedDataset)
+      )
```

Comment:
**suggestion** Rather than casting, we could check the presence of `configOverride` with `typeof childConstructorWithConfig.configOverride === 'function'` before adding it to the list of configs to merge, maybe?